management-core/src/test/java/com/ns/task/product/ProductCoreServiceImpTest.java

```diff
@@ -0,0 +1,101 @@
+package com.ns.task.product;
+
+
+import com.ns.task.model.ProductDTO;
+import com.ns.task.product.entities.ProductEntity;
+import com.ns.task.product.repositories.ProductRepository;
+import com.ns.task.product.service.ProductCoreServiceImpl;
+import com.ns.task.services.ProductService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.modelmapper.ModelMapper;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+
+@RunWith(SpringRunner.class)
+public class ProductCoreServiceImpTest {
+    @Mock
+    private ProductService productService;
+    @MockBean
+    private ProductRepository repository;
+    @Spy
+    private ModelMapper modelMapper;
+
+    @Before
+    public void setUp() throws Exception {
+        productService = new ProductCoreServiceImpl(modelMapper, repository);
+    }
+
+    @Test
+    public void getProducts() {
+        List<ProductEntity> productEntities = new ArrayList<>();
+        ProductEntity productEntity = new ProductEntity();
+        productEntity.setId(1);
+        productEntity.setName("SCREWDRIVER");
+        productEntity.setModel("S090");
+        productEntity.setPrice(new BigDecimal("12.2"));
+        productEntities.add(productEntity);
+        when(repository.findAll()).thenReturn(productEntities);
+        productService.getProducts();
+        assertEquals(productEntities.size(), 1);
```

Comment:
If you are going to verify just the size, I think you can omit all the setters for the entity, less code ;)

However, in this specific scenario, your service gets the data from DB and then returns as a different class, then I think is a good idea to test that.

Reply:
Yes, here I just trying to validate the mapping between Entity and DTO is working properly